src/game/card.h

```diff
@@ -0,0 +1,158 @@
+#pragma once
+
+#include <deque>
+#include <set>
+#include <algorithm>
+#include <random>
+
+#include <uno/uno.pb.h>
+
+namespace SailGame { namespace Game {
+
+using ::Uno::CardColor;
+using ::Uno::CardText;
+
+struct Card;
+
+using InitHandcardsT = std::array<Card, 7>;
+
+struct Card {
+    CardColor mColor;
+    CardText mText;
+
+    Card() = default;
+    Card(CardColor color, CardText text) : mColor(color), mText(text) {}
```

Comment:
const auto&,  by the way,  I don't see the definition of CardColor\ Text.
is it from generated pb?
is it enum? if it is enum,  just add const.

Reply:
they are from generated pb.
if it's passed by value, is `const` necessary?